test/interfaces/exchange_rate_provider_interface_test.rb

```diff
@@ -8,20 +8,13 @@ module ExchangeRateProviderInterfaceTest
   end
 
   test "exchange rate provider response contract" do
-    accounting_for_http_calls do
-      response = @subject.fetch_exchange_rate from: "USD", to: "MXN", date: Date.current
+    VCR.use_cassette "synth/exchange_rate" do
+      response = @subject.fetch_exchange_rate from: "USD", to: "MXN", date: Date.iso8601("2024-08-01")
 
       assert_respond_to response, :rate
       assert_respond_to response, :success?
       assert_respond_to response, :error
       assert_respond_to response, :raw_response
     end
   end
-
-  private
-    def accounting_for_http_calls
```

Comment:
@zachgoll great work here! In case you're wondering (you may have already figured this out) this was put in place to have a single point of extension to stub responses for all providers, since these tests will eventually be shared across them.

All good removing, and all good to never bring it back. Just realized it may not be obvious so thought I'd point it out.

Reply:
@josefarias ahh, good callout!  Was this the direction you were thinking?

```rb
def accounting_for_http_calls
  cassettes = [
   { name: 'synth/exchange_rate' },
   { name: 'other/exchange_rate' }
  ]
  
  VCR.use_cassettes(cassettes) do
    yield
  end
end
```

Reply:
Yes! Something like that